src/utils/validation.js

```diff
@@ -1,10 +1,61 @@
+import moment from 'moment';
+
 import store from '../data/store';
 import { jsonDeepCopy } from '.';
 import { courseSubmittingFailure } from '../data/actions/courseSubmitInfo';
-import { PUBLISHED } from '../data/constants';
+import { PUBLISHED, DATE_FORMAT } from '../data/constants';
 
 const requiredMessage = 'This field is required';
 
+// Dummy Validator: every input is valid
+const noValidate = () => undefined;
+
+// ensure that input is non-empty
+const requiredValidate = (value) => {
+  if (value !== undefined && value !== null && value !== '') {
+    return undefined;
+  }
+  return requiredMessage;
+};
+
+// ensure that the input is a valid datetime string in the format of DATE_FORMAT
+const dateTimeValidate = value => {
+  const dateTime = moment(value, DATE_FORMAT, true);
+  if (dateTime.isValid()) {
+    return undefined;
+  }
+  return 'Please enter a valid date';
+};
+
+const courseRunKeyValidate = value => {
+  const pattern = /^[a-zA-Z0-9-]*$/;
+  if (pattern.test(value)) {
+    return undefined;
+  }
+  return 'Enter a valid course key';
```

Comment:
nit: To make it consistent with other error messages, update this to `Please enter a valid course key`

Reply:
done